### PR TITLE
shell: backends: add init log level for dummy shell

### PR DESCRIPTION
**Pull request by marco85mars20**
Dummy shell was always started with log backend enabled with filter statically set to INFO level. AFAIK no use of such log backend can be made, thus causing waste of resources. The new changes keep INFO as the default filter level. In order to disable log backend to be attached to the Dummy shell, CONFIG_SHELL_DUMMY_INIT_LOG_LEVEL_NONE has to be set.

Signed-off-by: Marco Argiolas <marco.argiolas@ftpsolutions.com.au>

### CODE REVIEW

File: subsys/shell/backends/shell_dummy.c [hunk 1]
```diff
@@ -101,9 +101,16 @@ const struct shell_transport_api shell_dummy_transport_api = {
 static int enable_shell_dummy(const struct device *arg)
 {
 	ARG_UNUSED(arg);
+	bool log_backend = CONFIG_SHELL_DUMMY_INIT_LOG_LEVEL > 0;
+	uint32_t level = (CONFIG_SHELL_DUMMY_INIT_LOG_LEVEL > LOG_LEVEL_DBG) ?
+		      CONFIG_LOG_MAX_LEVEL : CONFIG_SHELL_DUMMY_INIT_LOG_LEVEL;
```
Comment (nixward) [inline subsys/shell/backends/shell_dummy.c:106]:
```suggestion
	uint32_t level = CONFIG_SHELL_DUMMY_INIT_LOG_LEVEL;
```
That check was redundant, KConfig logic will ensure the correct range.

**Comment by nixward** [comment]
_causing waste of resources_

Can you specify which resource please @marco85mars20 ?

**Comment by marco85mars20** [comment]
I was assuming some flash and RAM were used to support an extra log backend, but to my surprise, they are exactly matching with or without Log Backend enabled for the Dummy shell. Can you explain this?
I am going to further test the CPU load.

File: subsys/shell/backends/shell_dummy.c (hunk 1 above)
Comment (marco85mars20) [reply subsys/shell/backends/shell_dummy.c:106]: This is how they do in Zephyr, it's been copied over. I so this piece of code in multiple shells.

Comment (marco85mars20) [reply subsys/shell/backends/shell_dummy.c:106]: DEBUG is level 4, CONFIG_SHELL_DUMMY_INIT_LOG_LEVEL  can go up to 5, and it is not a valid log level.

Comment (nixward) [reply subsys/shell/backends/shell_dummy.c:106]: Ok that's strange I'll checkout why they have SHELL_DUMMY_INIT_LOG_LEVEL_DEFAULT ?

Comment (marco85mars20) [reply subsys/shell/backends/shell_dummy.c:106]: From the code, it looks like `SHELL_DUMMY_INIT_LOG_LEVEL_DEFAULT` has to be used when want the Log Backend to align with whatever has been set to be the highest log level (ie. `CONFIG_LOG_MAX_LEVEL`)

Comment (nixward) [reply subsys/shell/backends/shell_dummy.c:106]: Ah ok, nice! Disregard my original comment then.

**Comment by marco85mars20** [comment]
Decided to archive this because no significant improvement (RAM and CPU) has been noticed.

**Pull request closed without merging**